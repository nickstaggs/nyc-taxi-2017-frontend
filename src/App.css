.App {
  text-align: center;
  font-family: 'Gloria Hallelujah', cursive;
}

svg {
  stroke-width: 0.25px;
}

path:hover {
    stroke: 'white';
    stroke-width: 2px;
}

.selected {
  stroke: 'white';
  stroke-width: 2px;
}

.Manhattan {
    fill: red;
}

.Brooklyn {
    fill: blue;
}

.Queens {
    fill: green;
}

.Bronx {
    fill: yellow;
}

.Staten {
    fill: orange;
}

.EWR {
    fill: purple;
}

.floatRight {
  float: right;
}

.floatLeft {
  float: left;
}
