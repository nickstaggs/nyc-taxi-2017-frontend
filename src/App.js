import React from 'react';
import './App.css';
import Toolbar from './toolbar/Toolbar.js';
import Map from './map/Map.js';

const styles = {
  color: "blue"
}

class AppView extends React.Component {

  renderApp() {
    return (
      <div style={styles} className="App">
        <Toolbar {...this.props}/>
        <Map {...this.props} />
      </div>
    );
  }

  render() {
    return this.renderApp();
  }
}

class AppContainer extends React.Component {

  state = { pickupSelection: null, dropoffSelection: null, chloroplethData: null }

  updatePickupSelection = (zone) => {
    this.setState({ pickupSelection : zone });
  }

  updateDropoffSelection = (zone) => {
    this.setState({ dropoffSelection : zone });
  }

  updateChloroplethData = (data) => {
    this.setState({ chloroplethData : data });
  }

  render() {
    return <AppView {...this.state} updatePickupSelection={this.updatePickupSelection} 
                                    updateDropoffSelection={this.updateDropoffSelection} 
                                    updateChloroplethData={this.updateChloroplethData}/>;
  }
}

export default AppContainer;
