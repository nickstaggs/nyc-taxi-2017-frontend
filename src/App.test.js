import React from 'react';
import { shallow } from 'enzyme';
import App from './App';

it('view should render correctly without props', () => {
  const view = shallow(<App/>);

  expect(view).toMatchSnapshot();
});

