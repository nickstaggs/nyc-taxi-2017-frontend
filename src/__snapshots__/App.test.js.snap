// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`view should render correctly without props 1`] = `
<AppView
  chloroplethData={null}
  dropoffSelection={null}
  pickupSelection={null}
  updateChloroplethData={[Function]}
  updateDropoffSelection={[Function]}
  updatePickupSelection={[Function]}
/>
`;
